h2/src/main/org/h2/value/TypeInfo.java

```diff
@@ -341,7 +341,8 @@ public static TypeInfo getTypeInfo(int type, long precision, int scale, ExtTypeI
             }
         case Value.ENUM:
             if (extTypeInfo instanceof ExtTypeInfoEnum) {
-                return new TypeInfo(Value.ENUM, ValueEnum.PRECISION, 0, ValueEnum.DISPLAY_SIZE, extTypeInfo);
+                long enumPrecision = ((ExtTypeInfoEnum) extTypeInfo).getType().getPrecision();
+                return new TypeInfo(Value.ENUM, enumPrecision, 0, ValueEnum.DISPLAY_SIZE, extTypeInfo);
```

Comment:
Replace `ValueEnum.DISPLAY_SIZE` with `enumPrecision` too and remove `ValueEnumBase.DISPLAY_SIZE` field.

Reply:
Actually you can return `((ExtTypeInfoEnum) extTypeInfo).getType()` directly, there is no reason to create a new `TypeInfo` instance here.

Reply:
Both `ValueEnumBase.PRECISION` and `ValueEnumBase.DISPLAY_SIZE` should be deleted, they will not be used after these changes and their values are meaningless.

Reply:
Do you really think this is a 'Significant contribution'? ;)
I have posted the licence statement to the h2-database mailing list.

Reply:
> Do you really think this is a 'Significant contribution'? ;)

We lure you in with the easy stuff, and then we put you to work cranking out new features, so we just like to be prepared :-)